internal/codespaces/api/api.go

```diff
@@ -474,6 +474,74 @@ func (a *API) GetCodespacesMachines(ctx context.Context, repoID int, branch, loc
 	return response.Machines, nil
 }
 
+// TODO - should this be/is this in a more general API?
```

Comment:
Reminder to myself to get this resolved before completing 😄

Reply:
It's fine that this stays contained in this package and tailored to Codespaces. When we want to generalize this in the future, it's easy for us to extract this somewhere else. 👍